File: techila/examples/python/Features/custom_package/run_packagetest.py
# Copyright 2012-2016 Techila Technologies Ltd.

# This script contains the Local Control Code containing two function
# definitions.
# These functions can be used to create a Bundle containing the 'techilatest'
# and for creating a computational Project that will import the Bundle.
#
# Usage:
#
# execfile('run_packagetest.py')
# bundlename=create_package()
# result=run_packagetest(bundlename)
#
# Import the techila package
import techila

# The function used to store the 'techilatest' package in a Bundle.
def create_package():
    # Create the Bundle from the 'techilatest' package.
    bundlename = techila.bundleit('techilatest', all_platforms = True)

    print('package bundlename is \'%s\'' % bundlename) # Print the Bundle name

    # Return the name of the Bundle.
    return(bundlename)

# This function is used to create a computational Project that will import
# the Bundle containing the 'techilatest' package.
def run_packagetest(bundlename):
    results = techila.peach(funcname = 'packagetest_dist', # Function executed on Workers
                            params = ['<vecidx>','<param>'], # Input arguments to the executable function
                            files = ['packagetest_dist.py'], # File that will be evaluated at the start
                            imports = [bundlename], # Import the Bundle containing the 'techilatest' package
                            peachvector = [1, 2, 4, 8, 16], # Peachvector containing five integer elements
                            )

    # Display the results
    for row in results:
        print(row)

File: techila/examples/R/Features/custom_library/run_packagetest.r
# Copyright 2010-2013 Techila Technologies Ltd.

# This R-script contains the Local Control Code, which contains
# functions for creating a custom R library Bundle and for creating a
# computational Project in the Techila environment.

# Load the techila library
library(techila)

# This function will create a Bundle from the 'techilaTestPackage' by
# using the 'bundleit' function.
#
# Usage:
# packagename <- create_package()
create_package <- function() {

  # Create a Bundle from the 'techilaTestPackage' package
  packagename <- bundleit("techilaTestPackage",
                          allPlatforms = TRUE,
                          sdkroot = "../../../..")
  # Display the name of the Bundle
  print(paste("The name of the Bundle is:", packagename))
  # Return the name of the Bundle
  packagename
}

# This function will call the 'peach' function, which will create the
# Project.
#
# Usage: result <- run_packagetest(input, packagename)
# input = String
# Example:
# result <- run_packagetest("testvalue", packagename)
run_packagetest <- function(input, packagename) {

  result <- peach(funcname = "packagetest_dist", # Function that will be executed on Workers
                  params = list(input), # Input parameters for the executable function
                  files = list("packagetest_dist.r"), # Files that will be sourced on Workers
                  imports = list(packagename), # Import the bundle created by the bundleit function
                  peachvector = 1:1, # Set the number of Jobs to one (1)
                  sdkroot = "../../../.." # Location of the techila_settings.ini file
                 )
}
